config/config.go

```diff
@@ -207,6 +226,55 @@ func (c Configuration) New(
 	return tracer, closer, nil
 }
 
+// FromEnv uses environment variables to set the tracer's Configuration
+func (c *Configuration) FromEnv() (*Configuration, error) {
```

Comment:
I am wondering whether:
```go
func FromEnv() (*Configuration, error) {
```
would be better/cleaner. Then we don't have to care about overrides. It would also align more with java configuration (`static fromEnv`).

Reply:
Might be, yes. This means that people can't load the env vars into an existing object. Is that OK?

Reply:
They can create configuration from ENV and then adjust behaviour in code

Reply:
And we then have `SamplerConfigFromEnv()` / `ReporterConfigFromEnv()` / `...FromEnv()` ? Sounds good to me.